dsp/window/window_complex_test.go

```diff
@@ -0,0 +1,191 @@
+	},
+	{
+		name: "BlackmanNuttallComplex", fn: BlackmanNuttallComplex,
+		want: []float64{0.000363, 0.002885, 0.015360, 0.051652, 0.130567, 0.266629, 0.457501, 0.675215, 0.869392, 0.984644,
+			0.984644, 0.869392, 0.675215, 0.457501, 0.266629, 0.130567, 0.051652, 0.015360, 0.002885, 0.000363},
+	},
+	{
+		name: "FlatTopComplex", fn: FlatTopComplex,
+		want: []float64{-0.000421, -0.003687, -0.017675, -0.045939, -0.070137, -0.037444, 0.115529, 0.402051, 0.737755, 0.967756,
+			0.967756, 0.737755, 0.402051, 0.115529, -0.037444, -0.070137, -0.045939, -0.017675, -0.003687, -0.000421},
+	},
+}
+
+//testing table (gaussian window)
+// want the same value in imag part as in real part, so use one float64 for both
+var complexGausWindowTests = []struct {
+	name  string
+	sigma float64
+	want  []float64
+}{
+	{
+		name: "GaussianComplex (sigma=0.3)", sigma: 0.3,
+		want: []float64{0.003866, 0.011708, 0.031348, 0.074214, 0.155344, 0.287499, 0.470444, 0.680632, 0.870660, 0.984728,
+			0.984728, 0.870660, 0.680632, 0.470444, 0.287499, 0.155344, 0.074214, 0.031348, 0.011708, 0.003866},
+	},
+	{
+		name: "GaussianComplex (sigma=0.5)", sigma: 0.5,
+		want: []float64{0.135335, 0.201673, 0.287499, 0.392081, 0.511524, 0.638423, 0.762260, 0.870660, 0.951361, 0.994475,
+			0.994475, 0.951361, 0.870660, 0.762260, 0.638423, 0.511524, 0.392081, 0.287499, 0.201673, 0.135335},
+	},
+	{
+		name: "GaussianComplex (sigma=1.2)", sigma: 1.2,
+		want: []float64{0.706648, 0.757319, 0.805403, 0.849974, 0.890135, 0.925049, 0.953963, 0.976241, 0.991381, 0.999039,
+			0.999039, 0.991381, 0.976241, 0.953963, 0.925049, 0.890135, 0.849974, 0.805403, 0.757319, 0.706648},
+	},
+}
+
+func TestWindowsComplex(t *testing.T) {
+	const tol = 1e-6
+
+	//Input data
+	src := make([]complex128, 20)
+	for i := range src {
+		src[i] = complex(1, 1)
+	}
+
+	for _, test := range complexWindowTests {
+		t.Run(test.name, func(t *testing.T) {
+			// Copy the input since we are mutating it.
+			srcCpy := make([]complex128, len(src))
+			copy(srcCpy, src)
+			dst := test.fn(srcCpy)
+
+			if !equalApprox(dst, test.want, tol) {
+				t.Errorf("unexpected result for window function %q:\ngot:%v\nwant:%v", test.name, dst, test.want)
+			}
+		})
+	}
+}
+
+func TestGausWindowComplex(t *testing.T) {
+	const tol = 1e-6
+
+	//Input data
+	src := make([]complex128, 20)
+	for i := range src {
+		src[i] = complex(1, 1)
+	}
+
+	for _, test := range complexGausWindowTests {
+		t.Run(test.name, func(t *testing.T) {
+			// Copy the input since we are mutating it.
+			srcCpy := make([]complex128, len(src))
+			copy(srcCpy, src)
+			dst := GaussianComplex(srcCpy, test.sigma)
+
+			if !equalApprox(dst, test.want, tol) {
+				t.Errorf("unexpected result for window function %q:\ngot:%v\nwant:%v", test.name, dst, test.want)
+			}
+		})
+	}
+}
+
+// TestLanczosNan tests Lanczos on NaN condition
+// Lanczos NaN condition is k=(N-1)/2.
+// N=2k+1 which is odd number.
+func TestLanczosComplexNaN(t *testing.T) {
```

Comment:
To perform a single test on this case of Lanczos window, i need to make an additional member of testing case struct about window lenght:
var windowTests = []struct {
	name string
	fn   func([]float64) []float64
	want []float64
	winLen int
}{...}
or i can just do all tests on odd length windows. I prefer the first approach, because it is more explicitly. 
In this case i had a question: should i perform all tests on odd and even length case, or only with Lanczos window, because i know, that here could be a problem?